### DISCUSSION TITLE
Get partition from RunStatusSensorContext

### DISCUSSION THREAD

**Discussion by pdstrnadJC** [dagster-io/dagster#13283] [Q&A]
I have a sensor defined using `@run_status_sensor`:
```
@run_status_sensor(
    run_status=DagsterRunStatus.SUCCESS,
    request_jobs=[job2, job3],
    default_status=DefaultSensorStatus.RUNNING,
    monitored_jobs=[job1]
)
def fn(context: RunStatusSensorContext):
    ...
```

job1 is a partitioned job, and inside the decorated function `fn` I want to access the partition which job1 was processing. I don't see that information available in the `RunStatusSensorContext` unless I get it from the tags:

```
context.dagster_run.tags["dagster/partition"]
```
But that doesn't seem ideal and also a bit awkward since partition information is exposed more formally in other places (for example, in an asset I can do `context.asset_partition_key_for_output()`).

Is there a better way to do this besides looking at the tags? 

I'm on 1.1.8 but don't see any changes related to this in the most recent release (I was looking at `run_status_sensor_definition.py`).

**Comment by sryza** [comment]
tags are currently the best way to do this. If you file an issue on Github, we can try to prioritize adding a first-class property on `RunStatusSensorContext`. We would also be happy to review a PR if you'd be up for writing one.

**Comment by pdstrnadJC** [comment]
Sounds good, I can work on that. Let me clone the repo and read through the docs on contributing, etc.

**Comment by pdstrnadJC** [comment]
@sryza, I've made the changes but am having trouble with the tests. I created a new test and try to call `create_run_for_pipeline()` and then `submit_run()` on my partitioned job (very similar to what happens in test `test_run_status_sensor()`) - then I was going to use `evaluate_sensors()` and check the ticks to make sure sensors work and partition information is passed in the context to the sensor function. 

Problem is, I can't figure out how to make the pipeline and run partitioned. I end up getting this error since the job and assets are partitioned but the run is not partitioned:
```
dagster._core.errors.DagsterInvariantViolationError: Cannot access partition_key for a non-partitioned run
```

I guess I could use `execute_in_process()` instead since that allows partition to be specified, but I was trying to follow the existing pattern. 

Any suggestions? Thanks!

**Reply by sryza** [reply]
Hey @pdstrnadJC - thanks for working on this. I think this is how I would do it. Does this work?
```python
        run = instance.create_run_for_pipeline(
            foo_job,
            external_pipeline_origin=external_job.get_external_origin(),
            pipeline_code_origin=external_job.get_python_origin(),
            tags={"dagster/partition": "bla"},
        )
```

**Comment by pdstrnadJC** [comment]
Great, thanks - that works. I only knew that the partition key is available in the tags dict once the sensor fires, but didn't realize the partition key is passed in as a tag all the way upstream when the job runs.

So this brings up another question: I'm making partition key a first class field on `RunStatusSensorContext`, and when the context is constructed the partition key will be read from the tags provided by the pipeline run object (technically it's an object of type `DagsterRun`). But should partition key also be a first class field on `DagsterRun`? That would probably be a more invasive change, so just wanted to clarify.

**Reply by sryza** [reply]
Yeah I think you could definitely make a case that it should be a first class field on DagsterRun, though the change probably isn't worth the invasiveness
